splinepy/helpme/create.py

```diff
@@ -298,6 +298,306 @@ def revolved(
+        T = []
+        A = []
+        x_collection = []
+
+        # evaluating transformation matrices for each trajectory point
+        for i in range(len(par_value)):
+            # calculation according to NURBS Book, eq. 10.27
+            # tangent vector x on trajectory at parameter value i
+            x = traj.derivative([par_value[i]], [1])
+            x = (x / _np.linalg.norm(x)).ravel()
+            x_collection.append(x)
+
+            # projecting B_(i) onto the plane normal to x
+            B.append(B[i] - _np.dot(B[i], x) * x)
+            B[i + 1] = B[i + 1] / _np.linalg.norm(B[i + 1])
+
+            # defining y and z axis-vectors
+            z = B[i + 1]
+            y = _np.cross(z, x)
+
+            # array of transformation matrices from global to local coordinates
+            T.append(_np.vstack((x, y, z)))
+
+            # array of transformation matrices from local to global coordinates
+            A.append(_np.linalg.inv(T[i]))
+
+        # own procedure, if trajectory is closed and B[0] != B[-1]
+        # according to NURBS Book, Piegl & Tiller, 2nd edition, p. 483
+        if _np.array_equal(
+            traj.evaluate([[0]]), traj.evaluate([par_value[-1]])
+        ) and not _np.array_equal(B[0], B[-1]):
+            # reset transformation matrices
+            T = []
+            A = []
+            B_reverse = [None] * len(B)
+            B_reverse[0] = B[-1]
+            for i in range(len(par_value)):
+                # redo the calculation of B using x_collection from before
+                # according to NURBS Book, eq. 10.27
+                B_reverse[i + 1] = (
+                    B_reverse[i]
+                    - _np.dot(B_reverse[i], x_collection[i]) * x_collection[i]
+                )
+                B_reverse[i + 1] = B_reverse[i + 1] / _np.linalg.norm(
+                    B_reverse[i + 1]
+                )
+                # middle point between B and B_reverse
+                B_reverse[i + 1] = (B[i + 1] + B_reverse[i + 1]) * 0.5
+                B_reverse[i + 1] = B_reverse[i + 1] / _np.linalg.norm(
+                    B_reverse[i + 1]
+                )
+                # defining y and z axis-vectors
+                z = B_reverse[i + 1]
+                y = _np.cross(z, x_collection[i])
+
+                # array of transformation matrices from global to local coordinates
+                T.append(_np.vstack((x_collection[i], y, z)))
+
+                # array of transformation matrices from local to global coordinates
+                A.append(_np.linalg.inv(T[i]))
+
+            # check if the beginning and the end of the B-vector are the same
+            if not _np.allclose(B_reverse[0], B_reverse[-1], rtol=1e-3):
+                _log.warning(
+                    "Vector calculation is not exact due to the "
+                    "trajectory being closed in an uncommon way."
+                )
+
+        ### ROTATION MATRIX AROUND Y ###
```

Comment:
as discussed, there's no XYZ

Reply:
Changed it to e1,e2,e3 - pls check if you're fine with it